### PR TITLE
Add Service Account Config to Scorecard

### PR DESCRIPTION
**Pull request by theishshah**
Signed-off-by: Ish Shah <ishah@redhat.com>Signed-off-by: Ish Shah <ishah@redhat.com>

### CODE REVIEW

File: pkg/apis/scorecard/v1alpha3/configuration_types.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ type Configuration struct {
 
 	// Storage is the optional storage configuration
 	Storage Storage `json:"storage,omitempty" yaml:"storage,omitempty"`
+
+	// ServiceAccount is the optional way to set a non-default SA
```
Comment (joelanford) [inline pkg/apis/scorecard/v1alpha3/configuration_types.go:26]: Nit: maybe a little more detail here? Try to describe what it is/does as a primary concern, and then if necessary go into the "why" as another GoDoc paragraph. Something along the lines of the following is probably sufficient:
```suggestion
	// ServiceAccount is the service account under which scorecard tests are run. This field is optional. If left unset, the `default` service account will be used.
```

File: pkg/apis/scorecard/v1alpha3/configuration_types.go [hunk 2]
```diff
@@ -22,6 +22,9 @@ type Configuration struct {
 
 	// Storage is the optional storage configuration
 	Storage Storage `json:"storage,omitempty" yaml:"storage,omitempty"`
+
+	// ServiceAccount is the service account under which scorecard tests are run. This field is optional. If left unset, the `default` service account will be used.
```
Comment (jmrodri) [inline pkg/apis/scorecard/v1alpha3/configuration_types.go:26]: nit: I really dislike LONG comments. I usually wrap comments to make them easier to read. I will not block this PR for that just more for edification :)

**Review by jmrodri — APPROVED** [review]
/lgtm

**Review by rashmigottipati — APPROVED** [review]
/lgtm

**Pull request merged**